samples/bot-sso-adaptivecard/nodejs/bots/botSSOAdaptiveCard.js

```diff
@@ -0,0 +1,171 @@
+const AdaptiveCardResponse = require('../resources/adaptiveCardResponseJson.json');
+const AdaptiveCardWithSSO = require('../resources/AdaptiveCardWithSSOInRefresh.json');
+const Options = require('../resources/options.json');
+var ACData = require("adaptivecards-templating");
+
+class BotSSOAdativeCard extends TeamsActivityHandler {
+    constructor() {
+        super();
+
+        this.onMembersAdded(async (context, next) => {
+            //Send welcome message when app installed
+            await this.sendWelcomeMessage(context);
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+
+        this.onMessage(async (context, next) => {
+            const text = context.activity.text;
+
+            if (text) {
+                switch (text) {
+                    case "login":
+                        await context.sendActivity({ attachments: [this.getInitialAdaptiveCard(),] });
+                        break;
+
+                    case "PerformSSO":
+                        await context.sendActivity({ attachments: [this.getAdaptiveCardWithSSO()] });
+                        break;
+
+                    default:
+                        const message = "Please send 'login' for options";
+                        await context.sendActivity(message);
+                        break;
+                }
+            }
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+    }
+
+    /**
+    * Invokes when Action.Submit button performs onclick action. 
+    * @param {turnContext} context
+    */
+    async onInvokeActivity(context) {
+        if (context._activity.name == "adaptiveCard/action") {
+            if (context._activity.value == null) {
+                return null;
+            }
+
+            var value = context._activity.value;
+
+            if (value["action"] == null) {
+                return null;
+            }
+
+            var actiondata = value["action"];
+
+            if (actiondata["verb"] == null) {
+                return null;
+            }
+
+            // authToken and state are absent, handle verb
+            let verb = actiondata["verb"];
+
+            var authentication = null;
+
+            // When adaptiveCard / action invoke activity from teams contains token in response to sso flow from earlier invoke.
+            if (value["authentication"] != null) {
+                authentication = value["authentication"];
+            }
+
+            if (authentication == null) {
+                // authToken and state are absent, handle verb
+                switch (verb) {
+                    case "initiateSSO":
+                        //when token is absent in the invoke.
+                        return this.getInitialAdaptiveCard();
+                }
+            }
+            else {
+                // when token is present, send authentication token.
```

Comment:
change it to: send adaptive card with login successful information

Reply:
fixed